lint/package.go

```diff
@@ -181,17 +183,22 @@ func (p *Package) scanSortable() {
 	}
 }
 
-func (p *Package) lint(rules []Rule, config Config, failures chan Failure) {
+func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error {
 	p.scanSortable()
 	var wg sync.WaitGroup
 	for _, file := range p.files {
 		wg.Add(1)
 		go (func(file *File) {
-			file.lint(rules, config, failures)
+			err := file.lint(rules, config, failures)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, "Error during linting:", err)
+				os.Exit(1)
+			}
 			wg.Done()
 		})(file)
 	}
 	wg.Wait()
+	return nil
 }
```

Comment:
Please refactor to use

https://pkg.go.dev/golang.org/x/sync/errgroup#Go

https://pkg.go.dev/golang.org/x/sync/errgroup#Wait

_Originally posted by @ccoVeille in https://github.com/mgechev/revive/pull/1126#discussion_r1850932089_

Something like this

```go
func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error { 
          p.scanSortable() 
          var wg errgroup.WaitGroup 
          for _, file := range p.files { 
                  file := file // not sure if needed since Go 1.22, but I'm doubtful with a go routine
                  wg.Add(1) 
                  wg.Go(func() error { 
                         defer wg.Done()
                         file.lint(rules, config, failures) 
                         err := file.lint(rules, config, failures) 
                         if err != nil { 
                                 return  fmt.Errorf("Error during linting: %w", err)
                         }
                         return nil
                  })
          } 
          wg.Wait() 
         return nil 
  }
```

This is pseudo code written on a phone

Reply:
ok @ccoVeille  i made changes similar to https://pkg.go.dev/golang.org/x/sync/errgroup#example-Group-JustErrors